src/optimizer/statistics/stats_storage.cpp

```diff
@@ -6,39 +6,99 @@
 #include "loggers/optimizer_logger.h"
 
 namespace noisepage::optimizer {
-common::ManagedPointer<TableStats> StatsStorage::GetTableStats(catalog::db_oid_t database_id,
-                                                               catalog::table_oid_t table_id) {
-  StatsStorageKey stats_storage_key = std::make_pair(database_id, table_id);
+
+StatsStorageValue &StatsStorage::GetStatsStorageValue(const catalog::db_oid_t database_id,
+                                                      const catalog::table_oid_t table_id,
+                                                      catalog::CatalogAccessor *accessor) {
+  stats_latch_.LockShared();
```

Comment:
nit: maybe call this `stats_storage_latch_` to distinguish it from the latch for the individual table.

Reply:
Done